src/tests/ftest/launch.py

```diff
@@ -20,164 +20,471 @@
+            test_file["yaml"] = tmp_yaml
+
         else:
-            remaining_args = 2
-        addnl_tests = sys.argv[remaining_args:]
-    else:
-        printhelp()
+            display(
+                "ERROR: No yaml file exists for {}; excluding".format(
+                    test_file["py"]))
+            test_file["yaml"] = None
+    return test_files
 
-    # make it easy to specify a directory as a parameter later
-    test_directory = os.getcwd()
 
-    # setup some aspects of the environment
-    with open(os.path.join(os.path.dirname(os.path.realpath(__file__)),
-                           "../../../.build_vars.json")) as f:
-        envdata = json.load(f)
+def map_machines(machines):
+    """Sequentially map each specified machine name to a replacement key.
 
-    BINDIR = envdata['PREFIX'] + '/bin'
-    SBINDIR = envdata['PREFIX'] + '/sbin'
-    PATH = os.environ.get('PATH')
-    os.environ['PATH'] = BINDIR + ':' + SBINDIR + ':' + PATH
-    os.environ['DAOS_SINGLETON_CLI'] = "1"
-    os.environ['CRT_CTX_SHARE_ADDR'] = "1"
-    os.environ['CRT_ATTACH_INFO_PATH'] = envdata['PREFIX'] + '/tmp'
+    Args:
+        machines (list): list of machine names to map to replacement keys
 
-    # build a list of test classes
-    test_files = filelist(test_directory)
-    if not test_files:
-        printhelp()
+    Retrurns:
+        dict: a dictionary of machines indexed by their repalcement key
 
-    avocado = ' avocado run'
-    if sparse:
-        output_options = ' --html-job-result on'
-    else:
-        output_options = ' --show-job-log --html-job-result on'
+    """
+    machine_mapping = {}
+    for index, machine in enumerate(machines):
+        if len(machine) > 0:
+            # Sequentially map each specified machine name to a replacement
+            # key:
+            #   boro-[A-Z]
+            ascii_upper_code = 65 + index
+            if ascii_upper_code <= 91:
+                alpha_name = "boro-{}".format(chr(ascii_upper_code))
+                machine_mapping[alpha_name] = machine
+            else:
+                display(
+                    "WARNING: Unused machine name specified ({}): {}".format(
+                        ascii_upper_code, machine))
+    return machine_mapping
+
+
+def run_tests(test_files, sparse, tag_filter, prepare, rename):
+    """Run or display the test commands.
+
+    Args:
+        test_files (dict): a list of dictionaries of each test script/yaml file
+        sparse (bool):
+        tag_filter (str):
+        prepare (bool): if True dispaly the test command instead of running it
+        rename (bool): rename the avocado test results directory to incude the
+            test script name
+
+    Returns:
+        int: a sum of all the return codes of each 'avocado run' command
+
+    """
+    return_code = 0
+    avocado_logs_dir = None
+    if rename:
+        avocado_logs_dir = get_output(
+            "avocado config | grep logs_dir | awk '{print $2}'").strip()
```

Comment:
To simplify https://avocado-framework.readthedocs.io/en/69.0/Configuration.html?highlight=config#config-file-parsing-order I figured I would just let the avocado command handle it :)

Reply:
Oh, yes.  I am not at all suggesting replicating what Avocadao is doing.  I was just pondering if they have exported a python/module interface for doing the same thing without having to shell out from python, just to invoke python again.

My bet would be they have not exported any such interface, but was just suggesting it might be interesting to investigate if they have.